src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @flow
+ * @file ContentSharing Element
+ * @author Box
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import type { $AxiosError } from 'axios';
+import API from '../../api';
+import Internationalize from '../common/Internationalize';
+import ErrorMask from '../../components/error-mask/ErrorMask';
+import UnifiedShareModal from '../../features/unified-share-modal';
+import messages from '../../features/unified-share-modal/messages';
+import { convertItemResponse, convertUserResponse } from '../../features/unified-share-modal/utils/convertData';
+import {
+    CLIENT_NAME_CONTENT_SHARING,
+    FIELD_ALLOWED_INVITEE_ROLES,
+    FIELD_ENTERPRISE,
+    FIELD_HOSTNAME,
+    FIELD_ID,
+    FIELD_NAME,
+    FIELD_TYPE as FIELD_ITEM_TYPE,
+    FIELD_SHARED_LINK,
+    FIELD_SHARED_LINK_FEATURES,
+    TYPE_FILE,
+    TYPE_FOLDER,
+    FIELD_PERMISSIONS,
+    FIELD_EXTENSION,
+    FIELD_DESCRIPTION,
+} from '../../constants';
+import { contentSharingMessages, CONTENT_SHARING_ERRORS } from './messages';
+import type { ErrorResponseData } from '../../common/types/api';
+import type { ItemType } from '../../common/types/core';
+import type { item as itemFlowType } from '../../features/unified-share-modal/flowTypes';
+import type { ContentSharingItemAPIResponse, ContentSharingSharedLinkType } from './types';
+
+type ContentSharingProps = {
+    apiHost: string,
+    itemID: string,
+    itemType: ItemType,
+    language: string,
+    showFormOnly?: boolean,
+    token: string,
+};
+
+function ContentSharing(props: ContentSharingProps) {
+    const { apiHost, itemID, itemType, language, token } = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?ContentSharingSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorMessage, setErrorMessage] = useState<?Object>(null);
+
+    const api = new API({
```

Comment:
The API instance will be recreated every time the component re-renders. Does the prior instance need to be cleaned up? If so, should the instance be stored in a ref?

Reply:
Yes, good call. I stored the API in another `useEffect` call that only runs when there are updates to the API props.